pkg/api/unversioned/types.go

```diff
@@ -47,6 +49,12 @@ type TypeMeta struct {
 	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,2,opt,name=apiVersion"`
 }
 
+// Object
+type Object interface {
+	GetObjectKind() ObjectKind
+	DeepCopyObject() Object
```

Comment:
I'm fine with extending this interface to do this.

@kubernetes/sig-api-machinery @lavalamp @smarterclayton